### DISCUSSION TITLE
Examples or instructions for PSSCH sidelink transmission?

### DISCUSSION THREAD

**Discussion by T-recks** [srsran/srsRAN_4G#1310] [Q&A]
Are there any examples of LTE PSSCH/PSCCH, aka sidelink, transmission for srsRAN? I see [an example in the documentation](https://docs.srsran.com/projects/4g/en/latest/app_notes/source/cv2x/source/index.html) of using [pssch_ue](https://github.com/srsran/srsRAN_4G/blob/master/lib/examples/pssch_ue.c) as a sidelink receiver, but no examples or documentation involving transmission using srsRAN.

**Comment by Salex406** [comment]
@T-recks Hello
did you find any information about transmission?

**Reply by T-recks** [reply]
I did not find any examples.

Ultimately I also realized that the MAC layer for sidelink is not implemented here. For my particular use case I don't want to work directly at the PHY layer, so I am waiting for MAC layer sidelink support to be implemented in some open source 5G stack (I happen to know some people that are actively working on it in OAI).

**Reply by Salex406** [reply]
Have you tried this? 

https://gitlab.eurecom.fr/matzakos/LTE-D2D

as i understand it is already in the main branch of OAI

**Reply by T-recks** [reply]
LTE sidelink was specified in Rel-12. Both that repository and the main OAI `develop` branch only implement the LTE MAC layer up to Rel-10. The PHY layers in OAI are stated to be up to Rel-10 with a limited subset Rel-12 features (for example, the PSSCH). Thus, again, the PHY is available but not the MAC for D2D/sidelink.

At least, that is what's stated in the README notes regarding openair1, openair2, and openair3 directories.
